hibernate-core/src/main/java/org/hibernate/persister/entity/NamedQueryLoader.java

```diff
@@ -47,15 +47,15 @@ public NamedQueryLoader(String queryName, EntityPersister persister) {
 	}
 
 	@Override
-	public Object load(Serializable id, Object optionalObject, SharedSessionContractImplementor session, LockOptions lockOptions) {
+	public Object load(Serializable id, Object optionalObject, SharedSessionContractImplementor session, LockOptions lockOptions, Boolean readOnly) {
 		if ( lockOptions != null ) {
 			LOG.debug( "Ignoring lock-options passed to named query loader" );
 		}
-		return load( id, optionalObject, session );
+		return load( id, optionalObject, session, readOnly );
 	}
 
 	@Override
-	public Object load(Serializable id, Object optionalObject, SharedSessionContractImplementor session) {
+	public Object load(Serializable id, Object optionalObject, SharedSessionContractImplementor session, Boolean readOnly) {
```

Comment:
Probably needs an overloaded variant with the previous signature:

```java
public Object load(Serializable id, Object optionalObject, SharedSessionContractImplementor session) {
```

Reply:
thanks. The same reason as the above comment.

Reply:
again, the previous signature is a deprecated one and has been implemented as default interface method.

Reply:
Although deprecated, doing the change this way causes existing persisters and loaders to not compile, as shown by your very own changes in tests. As custom persisters may also exist in user code, this is undesirable.

If the new method is non-abstract, the old method may reuse it. However if the new method is abstract, it should be default and fall back to the old method unless overridden.

Just undo your changes in the unrelated tests and ensure it compiles.

Reply:
The new method accepts a new parameter (i.e. readOnly boolean value) which is not accessible from old method. So it seems old method can reuse the new method, but not vice versa. The compiling seems fine for as you can see, this PR is green. The old method is still provided (though a default one is provided to pass `null` value as the new parameter to the new method) and existing client code relying on it won't be broken, unless I were wrong.

Reply:
The old method is always there (by default it will route to the new method by passing 'null' as 'readOnly' parameter), so old code won't be broken and compiling should be fine. They simply lack the feature to pass 'readOnly' value.

Reply:
Correct. I updated. Thanks a lot for your precious feedback. I really appreciated it.